atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/ReloadingCqlClusterContainerTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+import static com.palantir.logsafe.testing.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.CassandraServersConfig;
+import com.palantir.atlasdb.cassandra.ReloadingCqlClusterContainer.CqlClusterFactory;
+import com.palantir.atlasdb.cassandra.backup.CqlCluster;
+import com.palantir.atlasdb.keyvalue.cassandra.async.client.creation.ClusterFactory.CassandraClusterConfig;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.refreshable.SettableRefreshable;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReloadingCqlClusterContainerTest {
```

Comment:
there's a fair amount of repeated stuff in this class (you have `EMPTY_SERVERS_CONFIG`, yet also use `ImmutableDefaultConfig` directly in one of the tests at least). Consider extracting out some helper methods to make this better.

Reply:
Extracted a whole bunch of things. I moved it mostly to setup, since even the reloadingCqlClusterContainer doesn't need test specific things to create.